asdf/asdf.py

```diff
@@ -109,17 +109,15 @@ def __init__(self, tree=None, uri=None, extensions=None, version=None,
             standard.
 
         """
+        self._extensions = []
+        self._extension_metadata = {}
+        self._process_extensions(extensions)
 
         if custom_schema is not None:
-            self._custom_schema = schema.load_custom_schema(custom_schema)
-            schema.check_schema(self._custom_schema)
+            self._custom_schema = schema.load_schema(custom_schema, self.resolver, True)
```

Comment:
This was moved after the call to self._process_extensions so that we could use our AsdfFile's resolver when loading the custom schema.